src/test/java/org/embulk/filter/column/TestJsonVisitor.java

```diff
@@ -397,7 +400,300 @@ public void visitArray_Columns()
+
+        MapValue visited = subject.visit("$['json1']", map).asMapValue();
+        assertEquals("{\"k_1\":[{\"k_1\":\"v\"}],\"k_2\":{\"k_2\":\"v\"}}", visited.toString());
+    }
+
+    // Because the bracket notation of double quotes converted to single quotes internally
+    // it can be mixed with the bracket notation of single quotes
+    @Test
+    public void visit_withSingleQuotesAndDoubleQuotes()
+    {
+        PluginTask task = taskFromYamlString(
+                "type: column",
+                "columns:",
+                " - {name: \"$['json1']['k_1']\", src: \"$['json1']['k.1']\"}",
+                " - {name: '$[\"json1\"][\"k_1\"][0][\"k_1\"]', src: '$[\"json1\"][\"k_1\"][0][\"k.1\"]'}",
+                " - {name: '$[\"json1\"][\"k_2\"]', src: '$[\"json1\"][\"k.2\"]'}",
+                " - {name: '$[\"json1\"][\"k_2\"][\"k_2\"]', src: '$[\"json1\"][\"k_2\"][\"k.2\"]'}");
+        Schema inputSchema = Schema.builder()
+                .add("json1", JSON)
+                .build();
+        JsonVisitor subject = jsonVisitor(task, inputSchema);
+
+        // {"k.1":[{"k.1":"v"}], "k.2":{"k.2":"v"}}
+        Value k1 = ValueFactory.newString("k.1");
+        Value k2 = ValueFactory.newString("k.2");
+        Value v = ValueFactory.newString("v");
+        Value map = ValueFactory.newMap(
+                k1, ValueFactory.newArray(ValueFactory.newMap(k1, v)),
+                k2, ValueFactory.newMap(k2, v));
+
+        MapValue visited = subject.visit("$['json1']", map).asMapValue();
+        assertEquals("{\"k_1\":[{\"k_1\":\"v\"}],\"k_2\":{\"k_2\":\"v\"}}", visited.toString());
+    }
+
+    @Test
+    public void visit_withComplexRename()
+    {
+        PluginTask task = taskFromYamlString(
+                "type: column",
+                "columns:",
+                " - {name: \"$.json1['k____1']\", src: \"$.json1['k.-=+1']\"}",
+                " - {name: \"$.json1['k____1'][0]['k____1']\", src: \"$.json1['k____1'][0]['k.-=+1']\"}",
+                " - {name: \"$['json1']['k_2']\", src: \"$['json1']['k.2']\"}",
+                " - {name: \"$['json1']['k_2']['k_2']\", src: \"$['json1']['k_2']['k.2']\"}");
+        Schema inputSchema = Schema.builder()
+                .add("json1", JSON)
+                .build();
+        JsonVisitor subject = jsonVisitor(task, inputSchema);
+
+        // {"k.1":[{"k.1":"v"}], "k.2":{"k.2":"v"}}
+        Value k1 = ValueFactory.newString("k.-=+1");
+        Value k2 = ValueFactory.newString("k.2");
+        Value v = ValueFactory.newString("v");
+        Value map = ValueFactory.newMap(
+                k1, ValueFactory.newArray(ValueFactory.newMap(k1, v)),
+                k2, ValueFactory.newMap(k2, v));
+
+        MapValue visited = subject.visit("$['json1']", map).asMapValue();
+        assertEquals("{\"k____1\":[{\"k____1\":\"v\"}],\"k_2\":{\"k_2\":\"v\"}}", visited.toString());
+    }
+
+    @Test
+    public void visit_withColumnNameIncludingSingleQuotes()
```

Comment:
We need a test for jsonpath escape rathe than yaml escape. Snameyaml is not what we've implemented.